java/arcs/core/storage/DirectStore.kt

```diff
@@ -92,7 +92,15 @@ class DirectStore<Data : CrdtData, Op : CrdtOperation, T> /* internal */ constru
     }
 
     override fun off(callbackToken: Int) {
-        return proxyManager.unregister(callbackToken)
+        proxyManager.unregister(callbackToken)
+        if (proxyManager.isEmpty()) {
+            close()
+        }
```

Comment:
Do we ensure that all DirectStores, when closed, are evicted from the cache? I don't want to end up with a situation where a listener is removed, the store closes, but then a new listener is added (which will then do nothing)

Reply:
This is not a hard close, it just kicks the Idle out of the Conflated Channel to free memory, if a new listener is added, the DirectStore should start working again.  The hard close will happen in the LruCacheMap.onEvict. I've been staging these into small PRs to try and avoid flakyness with too much complexity. 

I could add a special bit of code to the Cache that checks if the Store has been closed, and if it has, make it evict-on-get, e.g.  LruCacheMap.get() checks the cached value with a predicate, and if it is false, evicts and returns a fresh copy.

Reply:
Actually I take it back, it is a hard close because it closes the writeback. I'll add in a eviction code check.